src/v/cluster/rm_stm.cc

```diff
@@ -1362,6 +1362,14 @@ ss::future<result<kafka_result>> rm_stm::replicate_seq(
         co_return errc::invalid_request;
     }
 
+    ss::lw_shared_ptr<mutex> idempotent_lock = nullptr;
+    std::optional<ss::semaphore_units<ss::named_semaphore_exception_factory>>
+      units;
+    if (!bid.is_transactional) {
```

Comment:
we don't need to check `is_transactional` because we already did before dispatching the call to `replicate_seq` instead of `replicate_tx`